packages/filesystem/src/node/node-file-upload-service.ts

```diff
@@ -63,11 +64,16 @@ export class NodeFileUploadService implements BackendApplicationContribution {
         }
         try {
             const target = FileUri.fsPath(fields.uri);
-            await fs.move(request.file.path, target, { overwrite: true });
+            if (!fields.leaveInTemp) { // if dont move the file, just leave it where it is
+                await fs.move(request.file.path, target, { overwrite: true });
+            } else {
+                fs.rename(request.file.path, request.file.path.replace(request.file.filename, request.file.originalname));
+            }
```

Comment:
Minor: The comment should be on the rename path. Also we should await the `rename` operation so that it can be caught by the try-catch block.
```suggestion
            if (!fields.leaveInTemp) {
                await fs.move(request.file.path, target, { overwrite: true });
            } else {
                // leave the file where it is, just rename it to its original name
                await fs.rename(request.file.path, request.file.path.replace(request.file.filename, request.file.originalname));
            }
```